index-scheduler/src/utils.rs

```diff
@@ -538,3 +538,37 @@ impl IndexScheduler {
         }
     }
 }
+
+pub fn dichotomic_search(start_point: usize, mut is_good: impl FnMut(usize) -> bool) -> usize {
```

Comment:
I wanted to propose you use a generic "popular" crate to do this dichotomic search (and avoid logic mistakes), but I am astonished that [no single crate contains the word `dichotomic` at all](https://lib.rs/search?q=dichotomic) 😱